### DISCUSSION TITLE
Ways of displaying a local video in cross platform way

### DISCUSSION THREAD

**Discussion by 0xmemorygrinder** [tauri-apps/tauri#9235] [Q&A]
Hello,

I try creating an app using tauri which involves a video player for local assets in the middle of the window. I tried the streaming example in the beta version but since I am developping on linux I have a problem for loading the video and nothing happens (a white screen instead of the video playing).

From what I understand there is two main ways to achieve what I want : 
- Have the linux problem for video asset fixed (but I don't know how since it seems to involve webgtkkit)
- Create a child window embed in the main one and access the raw handle (again, I don't know if it is possible)  and perform the rendering by myself with a video library.

Can someone indicate me how should I proceed to get a local video playing in the window ?

Thanks in advance

**Comment by FabianLars** [comment]
Spawning a small axum or actix server in rust is likely easier than fixing webkitgtk or trying to inject a video player into the window. It's of course not ideal to serve files over localhost but 🤷

**Reply by 0xmemorygrinder** [reply]
Thanks for your answer.
What would be the better way to allow the user to select a file on their disk and play it in the window ?

**Reply by FabianLars** [reply]
You can use tauri's [dialog](https://tauri.app/v1/api/js/dialog/#open) (also avail in rust, or use rfd) and then use the file path in the request to the server (probably as part of the url if you want to use it as a <video> source)

**Reply by 0xmemorygrinder** [reply]
This is exactly what I did to select the file but how do I play it in tauri window ?
